core/libcamera_app.cpp

```diff
@@ -410,9 +410,9 @@ void LibcameraApp::StartCamera()
 		if (StillStream())
 			controls_.set(controls::FrameDurationLimits,
 						  libcamera::Span<const int64_t, 2>({ INT64_C(100), INT64_C(1000000000) }));
-		else if (options_->framerate > 0)
+		else if (!options_->framerate || options_->framerate.value() > 0)
 		{
-			int64_t frame_time = 1000000 / options_->framerate; // in us
+			int64_t frame_time = 1000000 / options_->framerate.value_or(30); // in us
```

Comment:
I guess the only slight feeling I get reading this is that there's quite a lot of `value_or(30)` creeping in around the place.  You might think, why  30? Could I use something else? And so on.

Maybe the `Options` class should have a `get_framerate()` method which does that? And the `framerate` and `framerate_` might be private? Or maybe that is just classic C++ over-engineering! I'm not sure... and in this case not all that bothered either, I'm fine with it as it is too.

Reply:
Yes, this bothers me a bit too!